### ISSUE TITLE
Some characters such as that for bool or nat are showing up as rectangles

### ISSUE THREAD

**Issue by Devagamster** [cpitclaudel/company-coq#18]
I have tried installing the suggest fonts and using them as my current font face, but no matter what I do, nat and bool character replacements just show as boxs. Is there a way to select what replacements I want and which ones I dont want?

**Comment by cpitclaudel** [comment]
Sure! <kbd>M-x customize-variable RET company-coq-prettify-symbols-alist RET</kbd> should do nicely.

**Comment by cpitclaudel** [comment]
Trying to troubleshoot the root issue too, though: did you try installing Symbola and adding `(set-fontset-font t 'unicode (font-spec :name "Symbola") nil 'append)` to your .emacs? And if that doesn't work, could you try `(set-fontset-font t 'unicode (font-spec :name "Symbola"))`?

**Comment by Devagamster** [comment]
Just tried both of those and no dice. I dont know if it helps at all but I'm using spacemacs as my starter kit and to manager my plugins, so that might be screwing with things. I can switch the font to fonts with good symbol support but it seems like they are just getting ignored. My default font is Adobe Source Code Pro if that helps at all... Also I'm running on windows which I know can mess some things up with emacs. Thanks for your help anyways :) For now Ill just remove the offending substitutions manually.

**Comment by cpitclaudel** [comment]
Hmm. That's unfortunate; but at least you get most of the symbols, right?

I wonder if explicitly selecting Symbola from the font selection menu solves it? (As in, it should turn everything into symbola, and then hopefully the symbols would appear). Another interesting thing would be to put the point on one of these boxes and press `C-u C-x =`; that should say which font is being used for rendering.

One common problem is that a font claims to support a particular character, but in fact does not. That can trip up the fallback mechanism.

**Comment by Devagamster** [comment]
Oh yeah most of them show up, just the bool, nat, and real substitutions show up as boxes. I also tried just setting all of the fonts to symbola but that didn't work either. I think it might have something to do with how Spacemacs handles fonts that is screwing things up, so its probably on my side.

I removed the offending characters from the list like you suggested with the customize variable dialog, so I don't have them in my list at the moment. If you would like, I could do some more investigation for debugging purposes, but at the moment it is all working and so I'm fine as is.

Thank you so much for the help!
